lib/_tls_common.js

```diff
@@ -52,6 +54,13 @@ function SecureContext(secureProtocol, secureOptions, context) {
   if (secureOptions) this.context.setOptions(secureOptions);
 }
 
+function validateKeyCert(value, type) {
+  if (typeof value !== 'string' && !(value instanceof Buffer))
```

Comment:
Ah, brilliant! Replacing the two check works perfectly. I'll change this over and add some additional tests.

Regarding the new error message, it is currently `['string', 'buffer']` and I'm thinking it should change to `['string', 'ArrayBuffer']` with perhaps `string` => `String`. Would you have an opinion on how that should look?